exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -1622,11 +1669,43 @@ func (b *Bitfinex) WsSendAuth(ctx context.Context) error {
 	return nil
 }
 
-// WsAddSubscriptionChannel adds a new subscription channel to the
-// WebsocketSubdChannels map in bitfinex.go (Bitfinex struct)
-func (b *Bitfinex) WsAddSubscriptionChannel(chanID int, channel, pair string) {
-	chanInfo := WebsocketChanInfo{Pair: pair, Channel: channel}
-	b.WebsocketSubdChannels[chanID] = chanInfo
+// WsAddSubscriptionChannel adds a confirmed channel subscription mapping from id to original params
+func (b *Bitfinex) WsAddSubscriptionChannel(chanID int, channel, symbol string) error {
+	assetType, pair, err := assetPairFromSymbol(symbol)
+	if err != nil {
+		return err
+	}
+
+	var c *stream.ChannelSubscription
+	s := b.Websocket.GetSubscriptions()
+	for i := range s {
+		if strings.EqualFold(s[i].Channel, channel) && s[i].Currency.Equal(pair) {
```

Comment:
Might need to distinguish between asset types as well

Reply:
Agreed.
Note however that the [existing CS Equal method](https://github.com/thrasher-corp/gocryptotrader/blob/master/exchanges/stream/websocket.go#L956-L960) does not.

Reply:
Ah, right. I didn't do this because there's no guarantee at all that they'll match.
I had conflicting use-cases of empty asset and asset.Spot.
I backed away from it as a result.

Tests are passing now, so it was an issue in my runtime.

I'll make the change as requested and then revisit if I have issues again.

Done 👍